src/Forum/Content/Index.php

```diff
@@ -58,9 +58,11 @@ public function __invoke(Document $document, Request $request)
         ];
 
         $apiDocument = $this->getApiDocument($request->getAttribute('actor'), $params);
+        $defaultRoute = app()->make('flarum.settings')->get('default_route');
 
         $document->content = $this->view->make('flarum.forum::frontend.content.index', compact('apiDocument', 'page'));
         $document->payload['apiDocument'] = $apiDocument;
+        $document->canonicalUrl = $defaultRoute === '/all' ? app()->url() : $request->getUri()->withQuery('');
```

Comment:
Please inject the URL generator and the settings repository instead of using `app()`.

Reply:
@franzliedke What about `app()->url()`? That has some more logic than just getting the forum url (for example, if `url` is an array). Duplicating that logic would be pointless, right ?

Reply:
No, all of that should be taken care of if you call `$this->url->to('forum')->base()`.

Reply:
Using `app()` means we're hiding the dependencies of our class deep within its methods' implementation, whereas injecting them means listing them very clearly right in the constructor signature.

Reply:
Done @franzliedke.